Document missing sync APIs on Deno Deploy (#515)
grammyjs/website: site/docs/plugins/i18n.md

// Continue reading to find out how to configure the instance.
const i18n = new I18n<MyContext>({
  defaultLocale: "en", // see below for more information
  // Load all translation files from locales/. (Not working in Deno Deploy.)
  directory: "locales",
});

> // Translation files loaded this way works on Deno Deploy, too.

Reviewer: ```suggestion
// Translation files loaded this way works in Deno Deploy, too.
```

// Continue reading to find out how to configure the instance.
const i18n = new I18n<MyContext>({
  defaultLocale: "en", // see below for more information

> 
>   // DOES NOT work in Deno Deploy
>   directory: "locales", // Load all translation files from locales/.

Reviewer: ```suggestion
  // Load all translation files from locales/. (Not working in Deno Deploy.)
  directory: "locales", 
```

Reviewer: Yes, it is.

});

> // uncomment to asynchronously load translations in Deno Deploy

Reviewer: ```suggestion
// Translation files loaded this way works on Deno Deploy, too.
```

type MyContext = Context & SessionFlavor<SessionData> & I18nFlavor;

> const bot = new Bot<MyContext>(""); // <-- put your bot token here
> 
> // Be awared, in Deno Deploy service,
> // we can only load fluent scripts after init I18n.

Reviewer: Unnecessary imho. The user already has initialized a bot, and the two comments are extra.

```suggestion
```

Reviewer 2: The 2 comment lines are my fault, I supposed to remove it... But if you remove the line 399 the script will be broken if sb try to run it. The `bot` instance is never initialized until this line, isn't it?

Reviewer: And that piece of code's goal is not that. It's mentioned how to initialize the bot in the above snippets.

Reviewer: I think they will understand if they RTFM from the beginning

Reviewer 2: If you insist.

Author: The existing example code snippets fall into two categories:
1) Short snippets. Illustrate a single thing only. One feature, one API call, one way of combining code, etc. These do not include imports and setup code.
2) Long snippets. Illustrate a setup, a complete bot, a minimal copy-paste-run example, etc. These do include imports and setup code.

Reviewer: Thanks for clarifying it the formal way, @KnorpelSenf!

Reviewer 3: For me, the same opinion as @ngdangtu-vn. It's not a snippet, it's a full code that include import statements, init session interfaces, init session, a few commands... and we are hidden something that someone will copy this and will not work.

I removed from all languages but 100% disagree, isn't a snippet, is a code to copy-paste and try

Full code:
```
import {
  Bot,
  Context,
  session,
  SessionFlavor,
} from "https://deno.land/x/grammy/mod.ts";
import { I18n, I18nFlavor } from "https://deno.land/x/grammy_i18n/mod.ts";

interface SessionData {
  __language_code?: string;
}

type MyContext = Context & SessionFlavor<SessionData> & I18nFlavor;

const i18n = new I18n<MyContext>({
  defaultLocale: "en",
  useSession: true, // whether to store user language in session

  // DOES NOT work in Deno Deploy
  directory: "locales",
});

// Translation files loaded this way works in Deno Deploy, too.
// await i18n.loadLocalesDir("locales");

// Remember to register `session` middleware before
// registering middleware of the i18n instance.
bot.use(
  session({
    initial: () => {
      return {};
    },
  }),
);

// Register the i18n middleware
bot.use(i18n);

bot.command("start", async (ctx) => {
  await ctx.reply(ctx.t("greeting"));
});

bot.command("language", async (ctx) => {
  if (ctx.match === "") {
    return await ctx.reply(ctx.t("language.specify-a-locale"));
  }

  // `i18n.locales` contains all the locales that have been registered
  if (!i18n.locales.includes(ctx.match)) {
    return await ctx.reply(ctx.t("language.invalid-locale"));
  }

  // `ctx.i18n.getLocale` returns the locale currently using.
  if ((await ctx.i18n.getLocale()) === ctx.match) {
    return await ctx.reply(ctx.t("language.already-set"));
  }

  await ctx.i18n.setLocale(ctx.match);
  await ctx.reply(ctx.t("language.language-set"));
});
```

const i18n = new I18n<MyContext>({
  defaultLocale: "en",
  useSession: true, // whether to store user language in session

  // DOES NOT work in Deno Deploy
  directory: "locales",
});

> // uncomment to asynchronously load translations in Deno Deploy

Reviewer: ```suggestion
// Translation files loaded this way works on Deno Deploy, too.
```

Reviewer 2: I don't understand there is `too` in the end sentence. It's not like the `directory` method work with Deno Deploy right?

Reviewer: The too means that it will work on Deno Deploy alongside other Deno environments.

Reviewer 2: I understand differently. I thought that there is other way that supposes to mention before that can work on Deno Deploy which we do not. Imho, the previous comment contains more meaning and clearer then this one. It gave a clear instruction on what to do and for what case.

const i18n = new I18n<MyContext>({
  defaultLocale: "en",
  useSession: true, // whether to store user language in session

  // DOES NOT work in Deno Deploy
  directory: "locales",
});

> // Translation files loaded this way works on Deno Deploy, too.

Reviewer: ```suggestion
// Translation files loaded this way works in Deno Deploy, too.
```

  }),
);

> // Register the i18n middleware after session middleware

Reviewer: This change is not related to the PR's goal

```suggestion
// Register the i18n middleware
```

Reviewer 2: I'm sorry, my bad.

i18n.loadLocalesDirSync("locales-2"); // sync version
```

> Note that certain environments require you to use the `async` version.

> > For example, Deno Deploy does not support synchronous file operations.

Reviewer 2: Should we add the reference link for other developers (https://deno.com/deploy/docs/runtime-fs)

Author: Can you perform any further changes in a subsequent PQ? As soon as something is ready for translation, we try not to alter the PQs anymore so that they can be translated more easily.

Reviewer 2: PQ as **reason**?

I just thought that giving a reference will make other developers understand it easily.
